Point_set_3/include/CGAL/Point_set_3.h

```diff
@@ -643,22 +703,35 @@ class Point_set_3
     for (std::size_t i = 0; i < m_base.size(); ++ i)
       m_indices[i] = indices[i];
 
-    // for (std::size_t i = 0; i < 10; ++ i)
-    //   std::cerr << m_indices[i] << " ";
-    // std::cerr << std::endl;
-
     // Sorting based on the indices reorders the point set correctly
     quick_sort_on_indices ((std::ptrdiff_t)0, (std::ptrdiff_t)(m_base.size() - 1));
 
-    // for (std::size_t i = 0; i < 10; ++ i)
-    //   std::cerr << m_indices[i] << " ";
-    // std::cerr << std::endl;
-
     m_base.resize (size ());
     m_base.shrink_to_fit ();
     m_nb_removed = 0;
   }
 
+  /*!
+    \brief Restores all removed points.
+    
+    After removing one or several points, calling this method restores
+    the point set to its initial state: points that were removed (and
+    their associated properties) are restored.
+
+    \note This method is only guaranteed to work if no point was
+    inserted after the removal: otherwise, some points might not be
+    restored.
+
+    \note If `collect_garbage()` was called after removal, the points
+    are irremediably lost and nothing will be restored.
+  */
+  void cancel_removal()
```

Comment:
Should it be a `cancel_removals`? When reading the function name I thought it was for only one point.

Reply:
In my mind, "cancel removal" was a general term (= "cancel the action of removing something"). But I agree that it can be misleading, I'll change for you version which is clearer.